source/posix/socket.c

```diff
@@ -1089,12 +1090,24 @@ int aws_socket_set_options(struct aws_socket *socket, const struct aws_socket_op
 
         if (socket->options.keep_alive_interval_sec && socket->options.keep_alive_timeout_sec) {
             int ival_in_secs = socket->options.keep_alive_interval_sec;
-            setsockopt(socket->io_handle.data.fd, IPPROTO_TCP, TCP_KEEPIDLE, &ival_in_secs, sizeof(ival_in_secs));
+            success =
```

Comment:
Should we warn on success != 0?

Reply:
How about I do bit ors on it and warn on the final step that one or more failed? Theres not a whole lot that can be done to fix it, and it's not the end of the world if this fails.